cmake/metal.14.0.diff

```diff
@@ -0,0 +1,36 @@
+diff -ur Metal/MTLEvent.hpp MetalNew/MTLEvent.hpp
+--- Metal/MTLEvent.hpp	2023-06-01 12:18:26
++++ MetalNew/MTLEvent.hpp	2024-04-15 07:36:59
```

Comment:
Just a comment on this patch for posterity. The `waitUntilSignaledValue` was retro-acitvely enabled recently (like a couple months ago) at least as far back as OS's that MLX supports. It has not percolated into metal-cpp yet. So I added the bindings here. However, we should remove these patches once it lands in metal-cpp.